src/libAtomVM/nifs.c

```diff
@@ -2931,6 +2939,21 @@ static term nif_erlang_erase_1(Context *ctx, int argc, term argv[])
     return old;
 }
 
+static term nif_erlang_memory(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    term type = argv[0];
+    VALIDATE_VALUE(type, term_is_atom);
+
+    if (type == BINARY_ATOM) {
+        size_t size = refc_binary_total_size(ctx);
+        return term_make_maybe_boxed_int64(ctx, size);
```

Comment:
We probably would want a ensure_free here for boxed ints, even if it's unlikely to be larger than 2^27 on 32 bits platforms or 2^59 on 64 bits.